src/util/dbid.cpp

```diff
@@ -0,0 +1,16 @@
+#include "util/dbid.h"
+
+
+const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
+
+//static
+DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
+    if (variant.canConvert<value_type>() &&
+            variant.convert(kVariantType)) {
+        value_type value = variant.value<value_type>();
```

Comment:
How about just using   "int toInt(bool *ok = 0) const" 

IMHO here is performance for beeing able to change the type in future (which will be unlikely to happen). 
This class is used very often thougout the Mixxx source. 

If you like to keep the flexibility, we may turn DbId into a template and implement  DbId<int>::valueOf() by toInt()